### ISSUE TITLE
snapshot error ld: library not found for -lPods-AppName (using Cocoapods 1.0.0.rc.2)

### ISSUE THREAD

**Issue by rubnov** [fastlane/fastlane#4594]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I am new to fastlane and snapshot. I followed the README and created a new UI testing target with a short swift file. Ran the test project from Xcode successfully (CMD+U). Also created a separate scheme for the UI tests. I am using Cocoapods 1.0.0.rc.2 with the following (example) podfile: 

```
platform :ios, '7.0'

abstract_target 'target' do

    pod 'somePod'
    pod 'anotherPod'

    target 'AppName'
    target 'AppNameUITests'
end
```

When running ```snapshot -s AppNameUITests``` I get the following error: 

❌  ld: library not found for -lPods-AppName
❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)

Building and running both AppName and AppUITests targets in Xcode works just fine. 
Running ```fastlane test``` yields the same result using the following Fastfile: 
 
Fastfile:
```ruby
# If you want to automatically update fastlane if a new version is available:
update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "1.86.1"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods
    
  end

  desc "Runs all the tests"
  lane :test do
    # scan works fine and I commented it out since I'm only interested in snapshot
    # scan(scheme: "PassengerUITests", device: "iPhone 6 (9.3)")
    snapshot
  end
 ...
end
```

Is this a known issue with Cocoapods 1.0? 
Can you help? 

### Environment

fastlane version 1.87.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? NO

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? NO

Attached the full fastlane log
[fastlane_log.txt](https://github.com/fastlane/fastlane/files/257351/fastlane_log.txt)

**Comment by rubnov** [comment]
Rolled back to Cocoapods 0.39 and everything seems to work fine, so it looks like Snapshot doesn't work with Cocoapods 1.0 yet?

**Comment by TKBurner** [comment]
@rubnov Thanks for following up. We don't test against 1.0 yet, so we might have some issues. However, your linker error suggests there is a problem linking a library against your test target. Can you have a look there? Thanks for working with me 👍

**Comment by KrauseFx** [comment]
This is not a `snapshot` issue. `snapshots just uses the `xcodebuild` command that is printed out when running. 

So when you run the UI Tests manually in Xcode everything works as expected? Did you make sure your testing scheme is marked as `Shared`?

**Comment by rubnov** [comment]
I finally figured out what was happening. 

While upgrading Cocoapods from 0.39 to 1.0, I changed my podfile in a way that caused cocoapods to generate a pods.a binary with a new name and link it with my target. The issue was that the old pods.a was still there and linked, too, against my target (Cocoapods did not clean this old reference from the build step). This created a conflict during linkage which generated the error. Not sure why or how Xcode itself did not complain when building the target from within the IDE. Yet, when I removed the old .a reference from the target's "link with libraries" build step the error was gone and I was able to build using fastlane.
